lib/libspl/include/sys/isa_defs.h

```diff
@@ -46,9 +46,13 @@ extern "C" {
 #define	__x86
 #endif
 
+#if defined(_ILP32)
+/* x32-specific defines; careful to *not* define _LP64 here */
+#else
 #if !defined(_LP64)
 #define	_LP64
 #endif
+#endif
```

Comment:
Even though x32 built kernels are rare, they do exist so let's make sure to add this same check to the `isa_defs.h` we use in the kernel `include/os/linux/spl/sys/isa_defs.h`.

Reply:
Oh neat, I didn't know you could build an x32 kernel, should try that!

Done